rero_ils/modules/stats/api.py

```diff
@@ -340,40 +340,31 @@ def get_librarian_library_pids(cls):
 
         Note: for system_librarian includes libraries of organisation.
         """
-        patron_search = PatronsSearch()\
-            .filter('term', user_id=current_user.id)\
-            .source(['libraries', 'roles', 'organisation']).scan()
-        patron_data = next(patron_search)
-
         library_pids = set()
-        if 'librarian' in patron_data.roles:
-            for library_pid in patron_data.libraries:
-                library_pids.add(library_pid.pid)
+        if 'librarian' in current_librarian["roles"]:
+            for library in current_librarian.get('libraries', []):
+                library_pids.add(extracted_data_from_ref(library))
 
         # case system_librarian: add libraries of organisation
-        if 'system_librarian' in patron_data.roles:
-            organisation_libraries_pid = LibrariesSearch()\
-                .filter('term',
-                        organisation__pid=patron_data.organisation.pid)\
+        patron_search = PatronsSearch()\
```

Comment:
I do not understand why you need to query the patron index to find the organisation. It should exists in the current_patron no?